src/api/app/controllers/webui/users/tokens_controller.rb

```diff
@@ -0,0 +1,84 @@
+class Webui::Users::TokensController < Webui::WebuiController
+  before_action :set_token, only: [:destroy]
+  before_action :set_params, :validate_package, :set_package, :set_scm_token, only: [:create]
+
+  after_action :verify_authorized, except: :index
+  after_action :verify_policy_scoped, only: :index
+
+  def index
+    @tokens = policy_scope([:webui, Token]).page(params[:page])
+  end
+
+  def new
+    @token = Token.new
+    authorize [:webui, @token]
+  end
+
+  def create
+    @token = Token.token_type(@params[:operation]).new(@params.merge(user: User.session!, package: @package, scm_token: @scm_token))
+
+    authorize [:webui, @token]
+
+    respond_to do |format|
+      format.js do
+        if @token.valid? && @token.save
+          render partial: 'create', locals: {
+            string: @token.string,
+            flash: { success: "Token successfully created! Make sure you save it - you won't be able to access it again." }
+          }
+        else
+          render partial: 'create', locals: {
+            flash: { error: "Failed to create token: #{@token.errors.full_messages.to_sentence}." }
+          }
+        end
+      end
+    end
+  end
+
+  def destroy
+    authorize [:webui, @token]
+    @token.destroy
+    flash[:success] = 'Token was successfully deleted.'
+    redirect_to tokens_url
+  end
+
+  private
+
+  def set_token
+    @token = Token.find(params[:id])
```

Comment:
I would move this code in `destroy` since it's only needed for this action. This would make the controller a tiny bit simpler with one less `before_action`.

Reply:
> I would move this code in `destroy` since it's only needed for this action. This would make the controller a tiny bit simpler with one less `before_action`.

In case we implement the update and trigger actions, it would be nice to have the `before_action` back.

Reply:
If it's going to be used in some other upcoming actions, then fine :+1:.